### PR TITLE
Switch Travis to Xenial build

### PR DESCRIPTION
**Pull request by adamretter**
I think this should now also run on Travis's new virtualised infrastructure which affords more memory and CPU.

We also need to think about migrating from travis-ci.org to travis-ci.com.

### CODE REVIEW

**Comment by siying** [comment]
Is Travis failure related?

**Comment by adamretter** [comment]
@siying possibly a problem with CMake. I just rebased and pushed again, let's see what happens...

**Comment by adamretter** [comment]
@siying @gfosco This one is ready to land now.

File: java/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,9 @@
 cmake_minimum_required(VERSION 3.4)
 
+if(${CMAKE_VERSION} VERSION_LESS "3.11.4")
+    message("Please consider switching to CMake 3.11.4 or newer")
```
Comment (tamird) [inline java/CMakeLists.txt]: You already error out in the relevant case below - is this warning helpful?

File: java/CMakeLists.txt [hunk 2]
```diff
@@ -348,20 +236,70 @@ add_jar(
...
+set(JAVA_HAMCR_JAR ${JAVA_TEST_LIBDIR}/hamcrest-core-1.3.jar)
+set(JAVA_MOCKITO_JAR ${JAVA_TEST_LIBDIR}/mockito-all-1.10.19.jar)
+set(JAVA_CGLIB_JAR ${JAVA_TEST_LIBDIR}/cglib-2.2.2.jar)
+set(JAVA_ASSERTJ_JAR ${JAVA_TEST_LIBDIR}/assertj-core-1.7.1.jar)
+set(JAVA_TESTCLASSPATH ${JAVA_JUNIT_JAR} ${JAVA_HAMCR_JAR} ${JAVA_MOCKITO_JAR} ${JAVA_CGLIB_JAR} ${JAVA_ASSERTJ_JAR})
+
+set(JNI_OUTPUT_DIR ${PROJECT_SOURCE_DIR}/java/include)
+file(MAKE_DIRECTORY ${JNI_OUTPUT_DIR})
+
+if(${Java_VERSION_MAJOR} VERSION_GREATER_EQUAL "10" AND ${CMAKE_VERSION} VERSION_LESS "3.11.4")
```
Comment (tamird) [inline java/CMakeLists.txt]: i believe any version 3.11 or higher is sufficient

**Comment by siying** [comment]
Let me land it now to cut the round trip time. Follow up comments can be addressed with follow up PR.

**Pull request closed without merging**